src/client/automation/playback/type/type-text.js

```diff
@@ -220,7 +220,7 @@ function _typeTextToContentEditable (element, text) {
     if (!startNode || !domUtils.isContentEditableElement(startNode) || !domUtils.isRenderedNode(startNode))
         return;
 
-    if (!simulateBeforeInput(element, text, browserUtils.isChrome))
+    if (!simulateBeforeInput(element, text, browserUtils.isChrome || browserUtils.isFirefox))
```

Comment:
Can we use the feature detection here (For example using the `nativeMethods.WindowInputEvent` property)?

Reply:
No, WindowInputEvent can have different input types: 'beforeinput' or 'input'. In previous FF versions the 'beforeinput' event was not supported, but WindowInputEvent was.
Morevover, there is a difference in event order in different browsers. The code above says that in Chrome and FF the `beforeinput` event should be raised before the `textInput` event. But there is also another code which raises the `beforeinput` event in Safari after the `textInput` event.
Thus, we cannot use feature detection here unfortunately.